pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -281,10 +323,52 @@ func manageRouteControllerManagerClientCA_v311_00_to_latest(client coreclientv1.
 	return resourceapply.SyncConfigMap(context.Background(), client, recorder, util.KubeAPIServerNamespace, apiserverClientCA, util.RouteControllerTargetNamespace, apiserverClientCA, []metav1.OwnerReference{})
 }
 
+// groupDisabledControllers gets caps from configv1.ClusterVersion and populates
+// groupDisabled with all controllers that are known in the ClusterVersion and
+// also disabled.
+func groupDisabledControllers(clusterVersion *configv1.ClusterVersion) map[controlplanev1.OpenShiftControllerName]bool {
+	knownCaps := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.KnownCapabilities...)
+	capsEnabled := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.EnabledCapabilities...)
+
+	groupDisabled := make(map[controlplanev1.OpenShiftControllerName]bool)
+	for _, cont := range knownControllers {
+		if controllerCapabilities[cont] != "" && knownCaps.Has(controllerCapabilities[cont]) && !capsEnabled.Has(controllerCapabilities[cont]) {
+			groupDisabled[cont] = true
+		}
+	}
+	return groupDisabled
+}
+
+// By default, all controllers are enabled. To disable specific controllers,
+// '-foo' disables the controller named 'foo'. For example, to disable the
+// OpenShiftServiceAccountController, one would use "-openshift.io/serviceaccount".
+func disableControllers(groupDisabled map[controlplanev1.OpenShiftControllerName]bool) []string {
+	controllers := []string{"*"}
+	for _, cont := range knownControllers {
+		if groupDisabled[cont] {
+			controllers = append(controllers, fmt.Sprintf("-%s", cont))
+		}
+	}
+	return controllers
+}
+
 // similar logic for route-controller-manager in manageRouteControllerManagerConfigMap_v311_00_to_latest
-func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
+func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(clusterVersionInformer configinformersconfigv1.ClusterVersionInformer, kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
```

Comment:
Let's make the informer a lister since the function only reads the version object. I.e. `clusterVersionLister configlisterv1.ClusterVersionLister` and `clusterVersionLister.Get("version")`. `configlisterv1 "github.com/openshift/client-go/config/listers/config/v1"`